corehq/sql_db/util.py

```diff
@@ -263,6 +263,16 @@ def filter_out_stale_standbys(dbs):
     ]
 
 
+def weighted_choice(choices, weights):
+    assert len(choices) == len(weights), 'Every choice needs a weight'
+    assert 0.9 < sum(weights) < 1.1, 'Weights should be normalized'
+
+    big_weights = (int(w * 100) for w in weights)
+    choices_lists = ([c] * w for c, w in zip(choices, big_weights))
+    weighted_by_count = [c for l in choices_lists for c in l]
```

Comment:
these two lines could be shortened to
```py
weighted_by_count = [c for c, w in zip(choices, big_weights) for i in range(w)]
```

Also add `from six.moves import range, zip` at the top of this file.

Reply:
Nice ... and that is elegant!